astropy/io/fits/hdu/image.py

```diff
@@ -945,7 +945,7 @@ def _calculate_datasum(self):
             else:
                 byteswapped = False
 
-            cs = self._compute_checksum(d.flatten().view(np.uint8))
```

Comment:
just to be clear, the difference between `ndarray.flatten` and `ndarray.ravel` is that the former returns a copy, while the latter doesn't, correct ?

Reply:
Yes, `ravel()` returns a view (unless it is not possible because the strides are not C aligned, then it makes a copy), while `flatten()` always makes a copy.